frontend/src/utils/cardButtons.js

```diff
@@ -0,0 +1,158 @@
+      for (let permission of TEAM_PERMISSIONS) result[permission] = true;
+    } else {
+      // we look at what teams this user belongs to. If authUser has a permission on one of those teams then they have the permission for the user
+      for (let authedTeamId in authUser.permissions.teams) {
+        if (viewedUser.teamMemberships[authedTeamId]) {
+          let heldPermissions =
+            authUser.permissions.teams[authedTeamId].permissions;
+          for (let permission of heldPermissions) {
+            result[permission] = true;
+          }
+        }
+      }
+    }
+    return result;
+  }
+  
+  export function getShowAddReviewButton({ card, permissions, isReviewer }) {
+    const REVIEW_STATUSES = [IN_REVIEW, COMPLETE, REVIEW_FEEDBACK];
+    return (
+      (isReviewer ||
+        permissions[REVIEW_CARDS] ||
+        permissions[TRUSTED_REVIEWER]
+      ) && REVIEW_STATUSES.indexOf(card.status) !== -1
+    )
+  }
+  
+  function getReviewRequestButtons({ card, permissions, isAssignee }) {
+    let showButtonRequestReview = false;
+    let showButtonCancelReviewRequest = false;
+    if (isAssignee || permissions[MANAGE_CARDS]) {
+      if (card.status === IN_PROGRESS) showButtonRequestReview = true;
+      if (card.status === REVIEW_FEEDBACK) showButtonRequestReview = true;
+      if (card.status === IN_REVIEW) showButtonCancelReviewRequest = true;
+    }
+  
+    return {
+      showButtonRequestReview,
+      showButtonCancelReviewRequest,
+    };
+  }
+  
+  export function showButtons({ card, authUser, viewedUser }) {
+    let showButtonStartProject = false;
+    let showButtonNoteWorkshopAttendance = false;
+    let showButtonCancelWorkshopAttendance = false;
+  
+    let showButtonStartTopic = false;
+    let showButtonStopTopic = false;
+    let showButtonEndTopic = false;
+  
+    let showButtonAddReview = false;
+    let showButtonRequestReview = false;
+    let showButtonCancelReviewRequest = false;
+  
+    const isAssignee = card.assignees.indexOf(authUser.userId) !== -1;
+    const isReviewer = card.reviewers.indexOf(authUser.userId) !== -1;
+    const permissions = getTeamPermissions({ authUser, viewedUser });
+  
+    let reviewRequestButtons;
+    if (card.contentTypeNice === "project") {
+      // PROJECT CARDS
+  
+      if (isAssignee && card.canStart) showButtonStartProject = true;
+      if (permissions[MANAGE_CARDS] & card.canForceStart)
+        showButtonStartProject = true;
+  
+      showButtonAddReview = getShowAddReviewButton({
+        card,
+        permissions,
+        isReviewer,
+      });
+      reviewRequestButtons = getReviewRequestButtons({
+        card,
+        permissions,
+        isAssignee,
+      });
+      showButtonRequestReview = reviewRequestButtons.showButtonRequestReview;
+      showButtonCancelReviewRequest =
+        reviewRequestButtons.showButtonCancelReviewRequest;
+    } else if (card.contentTypeNice === "workshop") {
+      // WORKSHOP CARDS
```

Comment:
this comment is flowtiing alone, like the one in line 80

You can either remove or move it on top of workshop cards